online/src/main/scala/ai/chronon/online/Fetcher.scala

```diff
@@ -190,23 +190,38 @@ class Fetcher(val kvStore: KVStore,
             val baseMap = internalMap ++ externalMap
             // used for derivation based on ts/ds
             val tsDsMap: Map[String, AnyRef] =
-              Map("ts" -> (requestTs).asInstanceOf[AnyRef], "ds" -> (requestDs).asInstanceOf[AnyRef])
-            val derivedMap: Map[String, AnyRef] = Try(
+              Map("ts" -> requestTs.asInstanceOf[AnyRef], "ds" -> requestDs.asInstanceOf[AnyRef])
+
+            val derivedMapTry: Try[Map[String, AnyRef]] = Try {
               joinCodec
                 .deriveFunc(internalResponse.request.keys, baseMap ++ tsDsMap)
                 .mapValues(_.asInstanceOf[AnyRef])
-                .toMap) match {
-              case Success(derivedMap) => derivedMap
-              case Failure(exception) => {
+            }
+
+            val (derivedMapCleaned, baseMapCleaned) = derivedMapTry match {
+              case Success(derivedMap) =>
+                val cleanedDerivedMap = derivedMap -- tsDsMap.keys
+                (cleanedDerivedMap, baseMap)
+              case Failure(exception) =>
                 ctx.incrementException(exception)
-                throw exception
-              }
+                val derivedExceptionMap =
+                  Map("derivation_fetch_exception" -> exception.traceString.asInstanceOf[AnyRef])
+                getJoinConf(joinName) match {
+                  case Failure(joinException) =>
+                    ctx.incrementException(joinException)
+                    (Map.empty[String, AnyRef], Map.empty[String, AnyRef])
+                  case Success(join) =>
+                    if (join.derivationsContainStar) {
+                      (derivedExceptionMap, baseMap)
+                    } else {
+                      (derivedExceptionMap, Map.empty[String, AnyRef])
+                    }
```

Comment:
this should just be 
```
(derivedExceptionMap, baseMap)
```

we want to keep `baseMap` even if derivations does not contain star.

Reply:
@nikhilsimha I feel like the same way Nikhil. Should we simply return baseMap always `(derivedExceptionMap, baseMap)`?

Reply:
If you check the downstream usage of `derivedMapCleaned` and `baseMapCleaned`: `
- `derivedMapCleaned` is returned back to user in responseMap, and thus is consistent to what users want to see
- `baseMapCleaned` is only used during logging.

Reply:
Thanks Haozhen. I renamed the variables and cleaned it up.